chain/x/zoracle/handler.go

```diff
@@ -23,13 +25,60 @@ func NewHandler(keeper Keeper) sdk.Handler {
 		case MsgEditOracleScript:
 			return handleMsgEditOracleScript(ctx, keeper, msg)
 		case MsgRequestData:
+			// TODO: Remove this hack!!!
+			// Here we assume that call data contains "sourceChannel + data"
+			// sourceChannel is always 10 characters
+			// sourceChannel := string(msg.Calldata[:10])
+			// calldata := hex.EncodeToString(msg.Calldata[10:])
+			// sourceChannelEnd, found := keeper.ChannelKeeper.GetChannel(ctx, "zoracle", sourceChannel)
+			// if !found {
+			// 	return nil, sdkerrors.Wrapf(sdkerrors.ErrUnknownRequest, "unknown channel %s port zoracle", sourceChannel)
+			// }
+			// destinationPort := sourceChannelEnd.Counterparty.PortID
+			// destinationChannel := sourceChannelEnd.Counterparty.ChannelID
+			// sequence, found := keeper.ChannelKeeper.GetNextSequenceSend(ctx, "zoracle", sourceChannel)
+			// if !found {
+			// 	return nil, sdkerrors.Wrapf(sdkerrors.ErrUnknownRequest, "unknown sequence number for channel %s port zoracle", sourceChannel)
+			// }
+
+			// fmt.Println(msg)
+			// packet := NewOracleRequestPacketData(
+			// 	msg.OracleScriptID, calldata, msg.RequestedValidatorCount,
+			// 	msg.SufficientValidatorCount, msg.Expiration, msg.PrepareGas,
+			// 	msg.ExecuteGas,
+			// )
+			// fmt.Println(packet.GetBytes())
+			// err := keeper.ChannelKeeper.SendPacket(ctx, channel.NewPacket(packet.GetBytes(),
+			// 	sequence, "zoracle", sourceChannel, destinationPort, destinationChannel,
+			// 	1000000000, // Arbitrarily high timeout for now
+			// ))
+			// if err != nil {
+			// 	return nil, err
+			// }
+			// return &sdk.Result{Events: ctx.EventManager().Events().ToABCIEvents()}, nil
 			return handleMsgRequestData(ctx, keeper, msg)
 		case MsgReportData:
 			return handleMsgReportData(ctx, keeper, msg)
 		case MsgAddOracleAddress:
 			return handleMsgAddOracleAddress(ctx, keeper, msg)
 		case MsgRemoveOracleAddress:
 			return handleMsgRemoveOracleAddress(ctx, keeper, msg)
+		case channeltypes.MsgPacket:
+			var requestData OracleRequestPacketData
```

Comment:
Why not use the same pattern
```
err := ...
if err != nil {
    return err...
}
```

Reply:
Because it's possible to try multiple types of packets?